### ISSUE TITLE
Large mod archives being locked upon finishing downloading via Vortex

### ISSUE THREAD

**Issue by Shiinachi** [Nexus-Mods/Vortex#7674]
Hey,

When downloading larger mod archives via Vortex, I'm frequently finding that they fail to install, claiming that Vortex doesn't have access to the file, or that it is in use by another program.

Following this, I typically try to delete the archive and redownload it. This is followed by a prompt asking me to give Vortex permissions to erase the old archive. This... then fails.

I chose to investigate this issue and found the archives within my mod download folder declared by Vortex. There's a problem though. The archives are compeltely locked out. They have no stated owner, and I cannot take ownership even as the in-built windows administrator. They cannot be deleted. They cannot be altered. The only conceivable way I could erase these files is by completely formatting my harddrive outwith Windows. See attached screenshots.

![image](https://user-images.githubusercontent.com/25741794/95799646-34599280-0ced-11eb-8550-c172608acc00.png)
![image](https://user-images.githubusercontent.com/25741794/95799672-4cc9ad00-0ced-11eb-9cc4-4ed0ef468414.png)

I possess no software which locks directories or files. To the best of my knowledge I have no sort of ransomware on my pc. My antivirus has not claimed this file, nor is it preventing me using it. Vortex has, for whatever reason, locked this file for me. I don't know whether files are locked at some point in the download process by Vortex, but as no other possible causes exist I can only conclude it does. No other files have been locked on my pc outwith the Vortex download location, nor have they had their file owner left empty. I've only noted this happening with larger files, 1GB and up. 

Restarting Vortex does not fix this issue, nor does running it with administrator privlages. Administrators are unable to access the file in any way. Attempting to force delete the file via command lines also does not work.
The only way I've been able to fix the ownership of these files so far is by restarting my PC entirely. This seems to fix the ownership issue.

I'm unsure if this is the best place to report this. As far as google searches go, this is not a frequent issue, either. 

Cheers.

**Comment by TanninOne** [comment]
You're making a bunch of assumptions that don't match your own observations.
> I possess no software which locks directories or files.

> The only way I've been able to fix the ownership of these files so far is by restarting my PC entirely.

These are contradictions. If restarting your PC unlocked the file then that **proves** that it was some software that had an exclusive lock on the file.
And this
> Restarting Vortex does not fix this issue,

**proves** that it was not Vortex.

> My antivirus has not claimed this file,

This is verifiably false, every AV has to lock files it wants to scan. It's possible that your AV is configured to not test vortex downloads or that it has already unlocked it by the time you checked. But considering you don't describe how you got to all those conclusions ("I possess no software which locks directories or files") I assume that you didn't check anything.

Here is a software that will actually test which application is locking a file: https://lockhunter.com/

**Comment by Shiinachi** [comment]
Based on the above, I went ahead and downloaded the tool you suggested.

Having just downloaded a mod for Fallout 4, I have experienced this issue again. See below for screenshots.

Error as reported by Vortex: 
![image](https://user-images.githubusercontent.com/25741794/97102389-ed7c7d00-169c-11eb-883b-6500a83eb11a.png)

File in question's security properties:
![image](https://user-images.githubusercontent.com/25741794/97102427-595ee580-169d-11eb-80a0-6817dda738e8.png)
It appears to have properly assigned me as the owner of the file after a moment.

Results as found by Lockhunter:
![image](https://user-images.githubusercontent.com/25741794/97102440-7398c380-169d-11eb-84dd-973ea316a161.png)


If I'm to guess, the file seems to be locked whilst downloading?
For whatever reason though, I can only assume that sometimes Vortex fails to unlock the files.

Rrelevant error log sections:
`Sun, 25 Oct 2020 08:29:33 GMT - debug: download finished (resumed) file=C:\Users\shiin\AppData\Roaming\Vortex\downloads\bladeandsorcery\fallout4\WIPAP - Worsins Immersive Power Armor Paints - Style 5 - v2.6 (Dirty)-36699-2-6-1575097244.7z
Sun, 25 Oct 2020 08:29:33 GMT - debug: setting final download name id=9dpthpG6K, fileName=WIPAP - Worsins Immersive Power Armor Paints - Style 5 - v2.6 (Dirty)-36699-2-6-1575097244.7z
Sun, 25 Oct 2020 08:29:33 GMT - debug: unfinished chunks chunks=[]
Sun, 25 Oct 2020 08:29:33 GMT - debug: downloading from address=195.181.173.81:443
Sun, 25 Oct 2020 08:29:33 GMT - debug: retrieving range id=45, range=bytes 74939330-128907139/236842755
Sun, 25 Oct 2020 08:30:09 GMT - debug: getting arguments from second instance 0=C:\Program Files\Black Tree Gaming Ltd\Vortex\Vortex.exe, 1=-d, 2=--allow-file-access-from-files, 3=--js-flags=--max-old-space-size=4096, 4=--original-process-start-time=13248088208570650, 5=nxm://fallout4/mods/33121/files/136145?key=X74O_P4eDe1uat-iXtM2MA&expires=1603787406&user_id=12765803
Sun, 25 Oct 2020 08:30:09 GMT - info: handling url url=nxm://fallout4/mods/33121/files/136145?key=X74O_P4eDe1uat-iXtM2MA&expires=1603787406&user_id=12765803
Sun, 25 Oct 2020 08:30:10 GMT - debug: new notification id=136145
Sun, 25 Oct 2020 08:30:10 GMT - debug: showing balloon title=Downloading from Nexus, content=Midwest Power Armor Evolution
Sun, 25 Oct 2020 08:30:10 GMT - info: queueing download goyKEGprI
Sun, 25 Oct 2020 08:30:14 GMT - debug: notification no longer exists 136145
Sun, 25 Oct 2020 08:30:19 GMT - debug: getting arguments from second instance 0=C:\Program Files\Black Tree Gaming Ltd\Vortex\Vortex.exe, 1=-d, 2=--allow-file-access-from-files, 3=--js-flags=--max-old-space-size=4096, 4=--original-process-start-time=13248088217898654, 5=nxm://fallout4/mods/15021/files/71407?key=RZibUh1o7bbrTTLj3uCsoQ&expires=1603787415&user_id=12765803
Sun, 25 Oct 2020 08:30:19 GMT - info: handling url url=nxm://fallout4/mods/15021/files/71407?key=RZibUh1o7bbrTTLj3uCsoQ&expires=1603787415&user_id=12765803
Sun, 25 Oct 2020 08:30:19 GMT - debug: new notification id=71407
Sun, 25 Oct 2020 08:30:19 GMT - debug: showing balloon title=Downloading from Nexus, content=Midwestern Power Armor Alt
Sun, 25 Oct 2020 08:30:19 GMT - info: queueing download vAQni-rHzO
Sun, 25 Oct 2020 08:30:23 GMT - debug: notification no longer exists 71407
Sun, 25 Oct 2020 08:30:28 GMT - info: start mod install id=WIPAP - Worsins Immersive Power Armor Paints - Style 5 - v2.6 (Dirty)-36699-2-6-1575097244
Sun, 25 Oct 2020 08:30:28 GMT - debug: got mod meta information archivePath=C:\Users\shiin\AppData\Roaming\Vortex\downloads\bladeandsorcery\fallout4\WIPAP - Worsins Immersive Power Armor Paints - Style 5 - v2.6 (Dirty)-36699-2-6-1575097244.7z, resultCount=1
Sun, 25 Oct 2020 08:30:28 GMT - debug: mod id for newly installed mod archivePath=C:\Users\shiin\AppData\Roaming\Vortex\downloads\bladeandsorcery\fallout4\WIPAP - Worsins Immersive Power Armor Paints - Style 5 - v2.6 (Dirty)-36699-2-6-1575097244.7z, modId=WIPAP - Worsins Immersive Power Armor Paints - Style 5 - v2.6 (Dirty)-36699-2-6-1575097244
Sun, 25 Oct 2020 08:30:28 GMT - debug: installing to modId=WIPAP - Worsins Immersive Power Armor Paints - Style 5 - v2.6 (Dirty)-36699-2-6-1575097244, destinationPath=D:\Vortex Mods\fallout4\WIPAP - Worsins Immersive Power Armor Paints - Style 5 - v2.6 (Dirty)-36699-2-6-1575097244
Sun, 25 Oct 2020 08:30:28 GMT - info: using install path id=WIPAP - Worsins Immersive Power Armor Paints - Style 5 - v2.6 (Dirty)-36699-2-6-1575097244, installPath=D:\Vortex Mods\fallout4\WIPAP - Worsins Immersive Power Armor Paints - Style 5 - v2.6 (Dirty)-36699-2-6-1575097244, fileName=WIPAP - Worsins Immersive Power Armor Paints - Style 5 - v2.6 (Dirty)-36699-2-6-1575097244
Sun, 25 Oct 2020 08:30:28 GMT - debug: extracting mod archive archivePath=C:\Users\shiin\AppData\Roaming\Vortex\downloads\bladeandsorcery\fallout4\WIPAP - Worsins Immersive Power Armor Paints - Style 5 - v2.6 (Dirty)-36699-2-6-1575097244.7z, tempPath=D:\Vortex Mods\fallout4\WIPAP - Worsins Immersive Power Armor Paints - Style 5 - v2.6 (Dirty)-36699-2-6-1575097244.installing
Sun, 25 Oct 2020 08:30:28 GMT - debug: extraction completed
Sun, 25 Oct 2020 08:30:28 GMT - warn: extraction reported error code=2, errors=ERROR: C:\Users\shiin\AppData\Roaming\Vortex\downloads\bladeandsorcery\fallout4\WIPAP - Worsins Immersive Power Armor Paints - Style 5 - v2.6 (Dirty)-36699-2-6-1575097244.7z
Can not open the file as archive

The process cannot access the file because it is being used by another process.

Sun, 25 Oct 2020 08:30:28 GMT - debug: removing temporary path D:\Vortex Mods\fallout4\WIPAP - Worsins Immersive Power Armor Paints - Style 5 - v2.6 (Dirty)-36699-2-6-1575097244.installing
Sun, 25 Oct 2020 08:30:28 GMT - info: finish mod install id=WIPAP - Worsins Immersive Power Armor Paints - Style 5 - v2.6 (Dirty)-36699-2-6-1575097244, outcome=undefined
Sun, 25 Oct 2020 08:30:28 GMT - error: install error message=Installation failed, details=The archive {{ installerPath }} is damaged and couldn't be installed. This is most likely fixed by re-downloading the file., installerPath=WIPAP - Worsins Immersive Power Armor Paints - Style 5 - v2.6 (Dirty)-36699-2-6-1575097244.7z
Sun, 25 Oct 2020 08:30:28 GMT - warn: Installation failed text=The archive {{ installerPath }} is damaged and couldn't be installed. This is most likely fixed by re-downloading the file., wrap=true
Sun, 25 Oct 2020 08:30:31 GMT - info: chunk completed id=43, numBuffers=16
Sun, 25 Oct 2020 08:30:31 GMT - debug: stopping chunk worker interrupted=false, id=43, offset=20971521, size=-1`

As far as other conditions go, my PC is typically running rather poorly whenever this issue occurs. After having Vortex open for extended periods of time, as in over 6 hours, and downloading 20+ mods, it typically starts running poorly in conjunction with resource-intensive programs like Chrome. Mayhaps that is interfering somehow in the unlocking process.

Whilst I can appreciate that many users come on here with baseless claims that Vortex has caused XYZ issue, I don't think treating every poster like they don't know what they're talking about is the best approach. I did appropriately test things before making the initial post. It was annoying me that 20GB+ of my PC was completely unaccessable. The fact I even tried enabling net admin accounts and using command prompts to override permissions should be proof of that. I'm posting this as a bug report, not a request for help. 

Hope this information helps. Cheers.
